### PR TITLE
Enforce Kapacitor alert schema agreement between client and server

### PR DESCRIPTION
**Pull request by goller**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### WIP TODOZ
  - [x] Add tests for go reversing of kapa
  - [x] Add tests for new reducers
  - [x] Add validation for rules that have functions but no every
  - [x] Fix which state is updated on rule create
  - [x] Test if null doesn't break rule updating

Connect #1531

### The problem
- In the Chronograf client, a user could not update an existing Kapacitor rule's field to use a function, if a function had not been applied when that rule was created.
- The client was including a default `30s` value for `.every` without respect to whether a field had yet been applied to the query (or a function to a field).
- The server was returning the Kapacitor rule request object to the client as the response, rather than the rule as it was actually persisted to the database.
- Server wasn't performing request validation (unlike other data coming to server)

### The solution
- Explicitly set the `.every` property of a rule to `30s` or `null` when adding a function to a field, or removing that field from the alert query, via dedicated Redux action creators.
- Return the persisted Kapacitor rule to the client as the response object.
- Have server perform the following minimum validation on requests:
  - has queryConfig
  - has every when there are functions

### Note
- There should probably be further validation performed by the server (https://github.com/influxdata/chronograf/issues/1545)

### CODE REVIEW

File: ui/src/kapacitor/reducers/rules.js [hunk 1]
```diff
@@ -45,6 +45,16 @@ export default function rules(state = {}, action) {
...
 
+    case 'ADD_EVERY': {
+      const {ruleID, frequency} = action.payload
+      return {...state, [ruleID]: {...state[ruleID], every: frequency}}
+    }
+
+    case 'REMOVE_EVERY': {
+      const {ruleID} = action.payload
+      return {...state, [ruleID]: {...state[ruleID], every: null}}
+    }
```
Comment (lukevmorris) [inline ui/src/kapacitor/reducers/rules.js]: These are nice and clean 👍

File: ui/src/kapacitor/containers/KapacitorRulePage.js [hunk 2]
```diff
@@ -23,6 +25,8 @@ export const KapacitorRulePage = React.createClass({
       loadDefaultRule: PropTypes.func.isRequired,
       fetchRule: PropTypes.func.isRequired,
       chooseTrigger: PropTypes.func.isRequired,
+      addEvery: PropTypes.func.isRequired,
```
Comment (lukevmorris) [inline ui/src/kapacitor/containers/KapacitorRulePage.js]: While we're here, can we do a PropTypes destructure above?

File: ui/src/kapacitor/components/DataSection.js [hunk 3]
```diff
@@ -27,6 +29,8 @@ export const DataSection = React.createClass({
       groupByTime: PropTypes.func.isRequired,
       toggleTagAcceptance: PropTypes.func.isRequired,
     }).isRequired,
+    onAddEvery: PropTypes.func.isRequired,
```
Comment (lukevmorris) [inline ui/src/kapacitor/components/DataSection.js]: While we're here, can we do a PropTypes destructure above?

File: ui/src/kapacitor/reducers/rules.js (hunk 1 above)
Comment (jaredscheib) [inline ui/src/kapacitor/reducers/rules.js]: All @goller there :) First iteration, too.

File: ui/src/kapacitor/containers/KapacitorRulePage.js (hunk 2 above)
Comment (jaredscheib) [inline ui/src/kapacitor/containers/KapacitorRulePage.js]: I don't think so. This should be tackled in a separate PR, along with the entire Kapacitor/Alerts app. We should consider using codeshift: https://github.com/reactjs/react-codemod

File: ui/src/kapacitor/components/DataSection.js (hunk 3 above)
Comment (jaredscheib) [inline ui/src/kapacitor/components/DataSection.js]: See above comment
